spec/features/openid_connect/openid_connect_spec.rb

```diff
@@ -136,30 +136,163 @@
     end
   end
 
-  it 'logout includes redirect_uris in CSP headers and destroys the session' do
-    id_token = sign_in_get_id_token
+  context 'when accepting id_token_hint in logout' do
+    before do
+      allow(IdentityConfig.store).to receive(:reject_id_token_hint_in_logout).
+        and_return(false)
+    end
 
-    state = SecureRandom.hex
+    context 'when sending id_token_hint' do
+      it 'logout includes redirect_uris in CSP headers and destroys the session' do
+        id_token = sign_in_get_id_token
 
-    visit openid_connect_logout_path(
-      post_logout_redirect_uri: 'gov.gsa.openidconnect.test://result/signout',
-      state: state,
-      id_token_hint: id_token,
-      prevent_logout_redirect: true,
-    )
+        state = SecureRandom.hex
 
-    current_url_no_port = URI(current_url).tap { |uri| uri.port = nil }.to_s
-    expect(current_url_no_port).to include(
-      "http://www.example.com/openid_connect/logout?id_token_hint=#{id_token}",
-    )
+        visit openid_connect_logout_path(
+          post_logout_redirect_uri: 'gov.gsa.openidconnect.test://result/signout',
+          state: state,
+          id_token_hint: id_token,
+          prevent_logout_redirect: true,
+        )
 
-    expect(page.response_headers['Content-Security-Policy']).to include(
-      'form-action \'self\' gov.gsa.openidconnect.test:',
-    )
+        current_url_no_port = URI(current_url).tap { |uri| uri.port = nil }.to_s
+        expect(current_url_no_port).to include(
+          "http://www.example.com/openid_connect/logout?id_token_hint=#{id_token}",
+        )
+
+        expect(page.response_headers['Content-Security-Policy']).to include(
+          'form-action \'self\' gov.gsa.openidconnect.test:',
+        )
+
+        visit account_path
+        expect(page).to_not have_content(t('headings.account.login_info'))
+        expect(page).to have_content(t('headings.sign_in_without_sp'))
+      end
+    end
+
+    context 'when permitting client_id' do
+      before do
+        allow(IdentityConfig.store).to receive(:accept_client_id_in_oidc_logout).
+          and_return(true)
+      end
+
+      context 'when sending client_id' do
+        it 'logout includes redirect_uris in CSP headers and destroys the session' do
+          sign_in_get_id_token
+          # see the definition of sign_in_get_token_response for where we define the
+          # client_id of the SP used
+          client_id = 'urn:gov:gsa:openidconnect:test'
```

Comment:
instead of repeating this comment, if we flip the order, it just becomes an explcit, self-documenting argument (ditto for the other spots we do this)
```suggestion
          client_id = 'urn:gov:gsa:openidconnect:test'
          sign_in_get_id_token(client_id: client_id)
```

Reply:
sweet, didn't realize `sign_in_get_id_token` took `client_id` as a parameter, this is much nicer!